src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/linkpicker/linkpicker.controller.js

```diff
@@ -92,9 +92,8 @@ angular.module("umbraco").controller("Umbraco.Editors.LinkPickerController",
                         $scope.anchorValues = resp.anchorValues;
                         $scope.model.target.url = resp.url;
                     });
-
                 }
-            } else if ($scope.model.target.url && $scope.model.target.url.length) {
+            } else if ($scope.model.target.url && $scope.model.target.url.length && !dialogOptions.hideAnchor) {
```

Comment:
This ensures the anchor is not populated from the URL when the anchor/query string field is hidden. Otherwise it's possible to enter a URL with an anchor/query string, submit and then not be able to edit this anymore (as it's split from the URL and that field is hidden).

An existing anchor value is kept however (it's only hidden after all), so this might result in duplicate anchor/query strings (the one in the URL, because it's not split anymore and the previously saved value).